File: verve patch/base/Applications/CSymphony_0_1/csSamples/CodeVariableReferenceExpression.cs
using System;
using System.CodeDom;
using System.CodeDom.Compiler;
using System.Collections;
using CSymphony.Lib;

namespace CSymphony.CSy
{
	public class CSyUnit : GraphUnit
	{
		public override void Build()
		{
			ns("MyNameSpace");

			declClass(TypePublic, "MyClass");
			VarRef testVar = "TestVariable";

			declMethod(MemPublic, "System.Void", "MyMethod");
			enter (testVar);

		}
	}
}

File: verve patch/base/Applications/CSymphony_0_1/vbSamples/CodeVariableReferenceExpression.vb
Option Strict Off
Option Explicit On

Imports System
Imports System.CodeDom
Imports System.CodeDom.Compiler
Imports System.Collections
Imports CSymphony.Lib

Namespace CSymphony.CSy
	Public Class CSyUnit
		 Inherits GraphUnit
		Overrides Public Sub Build()
			ns("MyNameSpace")

			declClass(TypePublic, "MyClass")

			Dim testVar As ArgRef = New ArgRef("TestVariable")

			declMethod(MemPublic, "System.Void", "MyMethod")
			
			enter (testVar.Expr())
		End Sub
	End Class
End Namespace
			